### PR TITLE
clarify the meaning of the weights term for TV denoising functions

### LINKED ISSUE #5379: clarify the meaning of the weights term for TV denoising functions
## Description
In the Zulip chat, there was [a question regarding the "weight" term](https://skimage.zulipchat.com/#narrow/stream/175598-general/topic/total.20variation.20documentation) in `denoise_tv_bregman`.

In that implementation, smaller weights lead to more smoothing whereas the opposite is true for `denoise_tv_chambolle`.

The underlying cause seems just to be which term in the cost function the weight applies to (i.e. either the data-consistency or the TV regularization term). We should update the docstrings to clarify where the weight is being applied to make its function less opaque.

### PR DESCRIPTION
**Pull request by gmelon**
## Description
Fixed  #5379 

After discovering above issue, I modified the docstring to clarify the meaning of weight a little more.
Please review it. Thanks you🙂

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by mkcor — COMMENTED** [review]
Thanks for your contribution, @gmelon.

As is, the only addition brought by this changeset is saying that 'more denoising' means 'more smoothing' -- it's not addressing #5379 (and since a noisy signal is notably not smooth, the reader already understands intuitively that 'more denoising' means 'more smoothing').

In the original issue, @grlee77 pointed out that what needs clarifying is which term in the cost function the weight applies to. And I would argue that the double negative form "the expense of less similarity to the `input`" calls for rewording.

**Comment by gmelon** [comment]
> Thanks for your contribution, @gmelon.
> 
> As is, the only addition brought by this changeset is saying that 'more denoising' means 'more smoothing' -- it's not addressing #5379 (and since a noisy signal is notably not smooth, the reader already understands intuitively that 'more denoising' means 'more smoothing').
> 
> In the original issue, @grlee77 pointed out that what needs clarifying is which term in the cost function the weight applies to. And I would argue that the double negative form "the expense of less similarity to the `input`" calls for rewording.

Thank you for the comments. In summary, do you mean that "the expense of less simplicity to the input" needs to be modified in some way?

**Comment by grlee77** [comment]
Ideally, we would add the actual cost functions using LaTeX math markup similarly to the example here:
https://github.com/scikit-image/scikit-image/blob/d52c11de0d27065dee177f87fd2bbf91e9b36660/skimage/metrics/simple_metrics.py#L196-L202

The user can then see the equation that is being minimized and where the `weight` occurs in it

Both functions are solving a similar problem, but the equation is formulated a bit differently in the two publications. If I recall correctly, in one of them the weight enforces similarity to the existing image whereas in the other the the weight is applied to the TV-regularization term in the equation. So, a larger `weight` in `denoise_tv_chambolle` gives a smoother image whereas for `denoise_tv_bregman` it is the opposite!

**Comment by grlee77** [comment]
I would use the following equation for `denoise_tv_chambolle`:

`\min_{(x \in X)} \sum_{i=0}^{N-1} \left| \nabla{x_i} \right| + \frac{1}{2\lambda}\sum_{i=0}^{N-1} \left ( u_i - x_i \right )^2`
which will render like:
eqn-white

The Bregman case can use the same equation, but with `lambda` in the numerator. 

In the above `u` is the noisy image and `x` is the denoised one. The first term enforces smoothness while the second enforces consistency with the original image, `u`.

**Comment by mkcor** [comment]
> > Thanks for your contribution, @gmelon.
> > As is, the only addition brought by this changeset is saying that 'more denoising' means 'more smoothing' -- it's not addressing #5379 (and since a noisy signal is notably not smooth, the reader already understands intuitively that 'more denoising' means 'more smoothing').
> > In the original issue, @grlee77 pointed out that what needs clarifying is which term in the cost function the weight applies to. And I would argue that the double negative form "the expense of less similarity to the `input`" calls for rewording.
> 
> @mkcor, Thank you for the comments. In summary, do you mean that "the expense of less simplicity to the input" needs to be modified in some way?

@gmelon no, I said *three* things, so they cannot be summarized into one. I said:

- [ ] the changes as of 891836b don't add any value, so I would suggest reverting them;
- [ ] the point is to clarify where the `weight` comes into play in the different functions to minimize; meanwhile, @grlee77 has provided much guidance on what to include;
- [ ] yes, I would suggest rewording the double negative form "the expense of less similarity to the `input`."

Please let us know if you run into any roadblocks!

**Comment by gmelon** [comment]
> Hi @gmelon,
> 
> thank you for your work on this PR so far! Would you like to continue?
> 
> If not, we can help finishing it.
> 
> Thanks again!

Thanks for asking!

I'm sorry to say that I can't continue this issue now.

Sorry again and Thanks you!
